server/util/body-parser.js

```diff
@@ -111,6 +112,56 @@ export function convertWpHeading(node) {
   }
 }
 
+function getWrappingTags(match) {
+  const maybeTag = match[1];
+  const isMatch = (maybeTag && maybeTag === match[2]);
+
+  return {
+    open: isMatch ? `<p><${maybeTag}>` : `<p>`,
+    close: isMatch ? `</${maybeTag}></p>` : `</p>`
+  };
+}
+
+function splitBlockquote(blockquote) {
+  // Match a string within quote marks and capture possible non-<p> wrapping tags
+  // Visualisation: https://regexper.com/#(%3F%3A%3C(%3F!p)(.*%3F)%3E)%3F%E2%80%9C%5B%5Cs%5CS%5D*%3F%E2%80%9D(%3F%3A.*%3C%5C%2F(%3F!p)(.*%3F)%3E)%3F
+  const quoteAndTagsRegex = new RegExp(/(?:<(?!p)(.*?)>)?“[\s\S]*?”(?:.*<\/(?!p)(.*?)>)?/);
+  const quoteAndTags = blockquote.match(quoteAndTagsRegex);
+
+  if (quoteAndTags) {
+    const wrapper = getWrappingTags(quoteAndTags);
+    const quote = blockquote.match(quoteAndTagsRegex)[0].replace(/[“”]/g, '');
```

Comment:
We're inside the `if (quoteAndTags)` conditional here, so accessing by key is safe. And we know that the Regex has two matching groups. So once we've checked that they're not undefined and they are equal, we're good.